virtual_rainforest/models/hydrology/hydrology_model.py

```diff
@@ -337,163 +355,202 @@ def update(self, time_index: int) -> None:
+                hydraulic_conductivity=self.constants.hydraulic_conductivity,  # m/s
+                hydraulic_gradient=self.constants.hydraulic_gradient,  # m/m
+                nonlinearily_parameter=self.constants.nonlinearily_parameter,
+                groundwater_capacity=self.constants.groundwater_capacity,
+                seconds_to_day=self.constants.seconds_to_day,
+            )
+            daily_lists["vertical_flow"].append(vertical_flow)
+
+            # Update soil moisture by +/- vertical flow to each layer and remove root
+            # water uptake by plants (transpiration), [mm]
+            soil_moisture_updated = update_soil_moisture(
+                soil_moisture=soil_moisture_evap,
+                vertical_flow=vertical_flow,
+                evapotranspiration=evapotranspiration,
+                soil_moisture_capacity=(
+                    self.constants.soil_moisture_capacity * soil_layer_thickness
+                ),
+                soil_moisture_residual=(
+                    self.constants.soil_moisture_residual * soil_layer_thickness
+                ),
+            )
+
+            daily_lists["soil_moisture"].append(
+                soil_moisture_updated / soil_layer_thickness
+            )
+
+            # update soil_moisture_mm for next day
+            soil_moisture_mm = soil_moisture_updated
+
+        # create output dict as intermediate step to not overwrite data directly
+        soil_hydrology = {}
 
-        # Calculate vertical flow in mm per time step for mean soil moisture
-        # Note that there are severe limitations to this approach on the temporal and
-        # spatial scale of this model and this can only be treated as a very rough
-        # approximation to discuss nutrient leaching.
-        # Further, we do not remove the water from the soil but assume a steady state
-        soil_hydrology["vertical_flow"] = calculate_vertical_flow(
-            soil_moisture=soil_moisture_evap,
-            soil_depth=soil_depth,
-            soil_moisture_capacity=self.constants.soil_moisture_capacity,
-            soil_moisture_residual=self.constants.soil_moisture_residual,
-            hydraulic_conductivity=self.constants.hydraulic_conductivity,
-            hydraulic_gradient=self.constants.hydraulic_gradient,
-            timestep_conversion_factor=self.constants.seconds_to_month,
-            nonlinearily_parameter=self.constants.nonlinearily_parameter,
-            meters_to_millimeters=self.constants.meters_to_millimeters,
+        # Calculate monthly accumulated values
+        for var in ["precipitation_surface", "surface_runoff", "soil_evaporation"]:
+            soil_hydrology[var] = DataArray(
+                np.sum(np.stack(daily_lists[var], axis=1), axis=1),
+                dims="cell_id",
+                coords={"cell_id": self.data.grid.cell_id},
+            )
+
+        soil_hydrology["vertical_flow"] = DataArray(
```

Comment:
Why is this one treated differently to the three previous variables?

Reply:
It is summed over a different axis because it has more dimensions (`layers` and `cell_id`) than the others (only `cell_id`)